examples/complete/main.tf

```diff
@@ -0,0 +1,152 @@
+locals {
+  validate_sm_region_cnd = var.existing_sm_instance_guid != null && var.existing_sm_instance_region == null
+  validate_sm_region_msg = "existing_sm_instance_region must also be set when value given for existing_sm_instance_guid."
+  # tflint-ignore: terraform_unused_declarations
+  validate_sm_region_chk = regex(
+    "^${local.validate_sm_region_msg}$",
+    (!local.validate_sm_region_cnd
+      ? local.validate_sm_region_msg
+  : ""))
+}
+
+
+##############################################################################
+# Resource Group
+##############################################################################
+
+module "resource_group" {
+  source = "git::https://github.com/terraform-ibm-modules/terraform-ibm-resource-group.git?ref=v1.0.5"
+  # if an existing resource group is not set (null) create a new one using prefix
+  resource_group_name          = var.resource_group == null ? "${var.prefix}-resource-group" : null
+  existing_resource_group_name = var.resource_group
+}
+
+
+##############################################################################
+# Key Protect All Inclusive
+##############################################################################
+
+
+module "key_protect_all_inclusive" {
+  providers = {
+    restapi = restapi.kp
+  }
+  source                    = "git::https://github.com/terraform-ibm-modules/terraform-ibm-key-protect-all-inclusive.git?ref=v3.1.2"
+  key_protect_instance_name = "${var.prefix}-kp"
+  resource_group_id         = module.resource_group.resource_group_id
+  region                    = var.region
+  resource_tags             = var.resource_tags
+  key_map                   = { "es" = ["${var.prefix}-es"] }
+  enable_metrics            = false
+}
+
+resource "ibm_iam_authorization_policy" "policy" {
+  source_service_name         = "messagehub"
+  source_resource_group_id    = module.resource_group.resource_group_id
+  target_service_name         = "kms"
+  target_resource_instance_id = module.key_protect_all_inclusive.key_protect_guid
+  roles                       = ["Reader"]
+}
```

Comment:
The auth policy creation need to move into the main modules code, and a variable called `skip_iam_authorization_policy` needs to be exposed to allow consumers to decide if they want it created or not (see [terraform-ibm-icd-postgresql](https://github.com/terraform-ibm-modules/terraform-ibm-icd-postgresql) as a reference)